File: Bots/JavaScript/Consumers/CodeFirst/WaterfallHostBot/skills/teamsSkill.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

const { ActivityEx } = require('botbuilder-core');
const { SkillDefinition } = require('./skillDefinition');

class TeamsSkill extends SkillDefinition {
  getActions () {
    return new Set([
      'TeamsTaskModule',
      'TeamsCardAction',
      'TeamsConversation',
      'Cards',
      'Proactive',
      'Attachment',
      'Auth',
      'Sso',
      'Echo',
      'FileUpload',
      'Delete',
      'Update'
    ]);
  }

  /**
   * @param {string} actionId
   */
  createBeginActivity (actionId) {
    if (!this.getActions().has(actionId)) {
      throw new Error(`[TeamsSkill]: Unable to create begin activity for "${actionId}".`);
    }

    // We don't support special parameters in these skills so a generic event with the right name
    // will do in this case.
    const activity = ActivityEx.createEventActivity();
    activity.name = actionId;

    return activity;
  }
}

module.exports.TeamsSkill = TeamsSkill;

File: Bots/Python/Consumers/CodeFirst/WaterfallHostBot/skills/teams_skill.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

from botbuilder.schema import Activity
from skills.skill_definition import SkillDefinition

SKILL_ACTION_TEAMS_TASK_MODULE = "TeamsTaskModule"
SKILL_ACTION_TEAMS_CARD_ACTION = "TeamsCardAction"
SKILL_ACTION_TEAMS_CONVERSATION = "TeamsConversation"


class TeamsSkill(SkillDefinition):
    def get_actions(self):
        return [
            SKILL_ACTION_TEAMS_TASK_MODULE,
            SKILL_ACTION_TEAMS_CARD_ACTION,
            SKILL_ACTION_TEAMS_CONVERSATION,
        ]

    def create_begin_activity(self, action_id: str):
        activity = Activity.create_event_activity()

        if action_id == SKILL_ACTION_TEAMS_TASK_MODULE:
            activity.name = SKILL_ACTION_TEAMS_TASK_MODULE

        elif action_id == SKILL_ACTION_TEAMS_CARD_ACTION:
            activity.name = SKILL_ACTION_TEAMS_CARD_ACTION

        elif action_id == SKILL_ACTION_TEAMS_CONVERSATION:
            activity.name = SKILL_ACTION_TEAMS_CONVERSATION

        else:
            raise Exception(
                f'[TeamsSkill]: Unable to create begin activity for "{ action_id }".'
            )

        return activity
